notebook/tests/selenium/test_buffering.py

```diff
@@ -0,0 +1,59 @@
+"""Tests buffering of execution requests."""
+
+from .utils import wait_for_selector
+
+
+def get_cell_text_output(notebook, index):
+    cell = notebook.cells[index]
+    output = wait_for_selector(cell, ".output_text", single=True)
+    return output.text
+
+
+def wait_for_kernel_ready(notebook):
+    wait_for_selector(notebook.browser, ".kernel_idle_icon")
+
+
+def kernels_buffer_without_conn(notebook):
+    """Test that execution request made while disconnected is buffered."""
+    # Assert that cell executed while kernel is disconnected still
+    # executes when reconnected
+    wait_for_kernel_ready(notebook)
+    notebook.browser.execute_script("IPython.notebook.kernel.stop_channels();")
+    notebook.edit_cell(index=0, content="print(1 + 2)")
+    notebook.execute_cell(0)
+    notebook.browser.execute_script("IPython.notebook.kernel.reconnect();")
+    wait_for_kernel_ready(notebook)
+    assert get_cell_text_output(notebook, 0) == "3"
+    notebook.delete_cell(0)
+
+
+def buffered_cells_execute_in_order(notebook):
+    """Test that buffered requests execute in order."""
+    notebook.append('k=1')
+    notebook.append('k+=1')
+    notebook.append('k*=3')
+    notebook.append('print(k)')
```

Comment:
The `.append()` method can take multiple strings at once to add a number of cells, if you wanted to shorten this a bit.